### ISSUE TITLE
Is the usage of scheduler described in README correct?

### ISSUE THREAD

**Issue by tamuhey** [huggingface/transformers#1898]
https://github.com/huggingface/transformers/blame/master/README.md#L546

I think it's not here to do `scheduler.step` but in epoch loop.
